Sources/DatadogSDKTesting/DDTestMonitor.swift

```diff
@@ -212,80 +212,107 @@ internal class DDTestMonitor {
+                    Log.print("ITR requires Git but Git Upload failed. Disabling ITR")
+                    itrBackendConfig = nil
                     return
                 }
-                let wildcardBranches = excludedBranches.filter { $0.hasSuffix("*") }.map { $0.dropLast() }
-                for branch in wildcardBranches {
-                    if currentBranch.hasPrefix(branch) {
-                        Log.debug("Excluded branch with wildcard: \(currentBranch)")
-                        itr = nil
-                        return
-                    }
-                }
+                itrBackendConfig = getItrConfig("Get ITR Config after git upload")
+                Log.debug("ITR config: \(String(describing: itrBackendConfig))")
             }
-
-            // Activate Intelligent Test Runner
-            if DDTestMonitor.config.itrEnabled && itrBackendConfig?.testsSkipping ?? false {
-                Log.debug("ITR Enabled")
-
-                if DDTestMonitor.config.applicationKey == nil {
-                    Log.print("APPLICATION_KEY env variable is not set, this is needed for Intelligent Test Runner")
-                    return
+        }
+        itrWorkQueue.addOperation(updateItrSettings)
+        
+        if DDTestMonitor.config.itrEnabled {
+            let isExcluded = { (branch: String) in
+                let excludedBranches = DDTestMonitor.config.excludedBranches
+                if excludedBranches.contains(branch) {
+                    Log.debug("Excluded branch: \(branch)")
+                    return true
                 }
-                
-                gitUploadQueue.waitUntilAllOperationsAreFinished()
-                guard isGitUploadSucceded else {
-                    Log.print("IRT is enabled but Git Upload failed. Disabling ITR")
-                    return
+                let match = excludedBranches
+                    .filter { $0.hasSuffix("*") }
+                    .map { $0.dropLast() }
+                    .first { branch.hasPrefix($0) }
+                if let wildcard = match {
+                    Log.debug("Excluded branch: \(branch) with wildcard: \(wildcard)*")
+                    return true
                 }
-                
-                itr = IntelligentTestRunner(configurations: DDTestMonitor.baseConfigurationTags)
-                itr?.start()
-            } else {
-                Log.debug("ITR Disabled")
-                itr = nil
+                return false
+            }
+            
+            if !isExcluded(branch) {
+                let itrSetup = BlockOperation { [self] in
+                    // Activate Intelligent Test Runner
+                    if itrBackendConfig?.testsSkipping ?? false {
+                        Log.debug("ITR Enabled")
+
+                        if DDTestMonitor.config.applicationKey == nil {
+                            Log.print("APPLICATION_KEY env variable is not set, this is needed for Intelligent Test Runner")
```

Comment:
application key is no longer needed for ITR endpoints - but this can be addressed at a different PR

Reply:
yes. This is a next task and PR